### PR TITLE
Update dev-dependencies

### PR DESCRIPTION
**Pull request by remcohaszing**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

- TypeScript options have been adjusted to TypeScript 5.
- `Array<void>` is not a valid type for `unified.Plugin`. it has been changed to an empty tuple.
- Since `@types/node` now exposes a global `URL`, all `URL` imports were removed.
- The missing dependency `@types/node` was added.
- XO was updated and issues autofixed.
- All imports from `vscode-languageserver-protocol` were changed to `vscode-languageserver`, meaning we don’t need it as an explicit dev dependency either.
- A workaround in a test has been removed, because the issue causing it was resolved upstream.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,21 +43,21 @@
     "vscode-languageserver-textdocument": "^1.0.0"
   },
   "devDependencies": {
-    "@types/tape": "^4.0.0",
+    "@types/node": "*",
```
Comment (ChristianMurphy) [inline package.json:46]: Would it make sense to pin the current LTS major version range?
Currently `^20.0.0`

**Review by ChristianMurphy — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (remcohaszing) [reply package.json:46]: For dependencies I think `*` makese sense. This allows the package manager to better deduplicate it.

For dev dependencies the same argument could be used, but it’s less important. Also `*` will show when the types break for some reason due to an updated.

An argument for using `^20.0.0` would be that it’s less likely to break builds. This is pretty much the same argument as for using a lockfile.

Another take could be to use the lowest supported Node.js version (`^14.0.0` in this case) to ensure backwards compatibility. However, that doesn’t support `node:test` yet.

I don’t really think there’s a right or wrong here.

Comment (ChristianMurphy) [reply package.json:46]: I'm okay leaving it.
I still have a mild preference for setting a range, for similar reasons to those you list.
But it's a non blocker.

Comment (wooorm) [reply package.json:46]: Hmm, I also don’t want to make a point of it here.
It’s a project Remco mostly works on, so feel free to do things the way that works for you!

But I’d personally don’t do it, and I’d probably vote against introducing it everywhere else.

> This allows the package manager to better deduplicate it.

I have my doubts. Other tools are going to specify versions, probably. I don’t think npm is running into problems understand the difference between `^20` and `*`

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
FYI I left a comment on a previously resolved subthread!

**Pull request merged**